### PR TITLE
Dynamic Go versions in vuln workflow

### LINKED ISSUE #2573: .github/workflows: use `https://go.dev/dl/?mode=json` to get Go versions for vuln.yml
### Operating System

- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] FreeBSD
- [ ] OpenBSD
- [ ] Android
- [ ] iOS
- [ ] Nintendo Switch
- [ ] Xbox
- [ ] Web Browsers

### What feature would you like to be added?

https://stackoverflow.com/questions/46623153/how-do-download-the-latest-version-of-go

> The problem with using curl https://golang.org/VERSION?m=text is that on some days, it will give "bleeding edge" result that is not actually available for download. Like today 2022-04-12 (: It returns 1.18.1 which is not yet available for download.

### Why is this needed?

_No response_

### PR DESCRIPTION
**Pull request by Skenvy**
# What issue is this addressing?
Closes #2573

## What _type_ of issue is this addressing?
refactor

## What this PR does | solves
Swap from hard coded versions of go to dynamically acquiring the latest stable versions from https://go.dev/dl/?mode=json

### CODE REVIEW

**Comment by Skenvy** [comment]
If not specifically avoiding participating in hacktoberfest, could this PR be labelled with "hacktoberfest-accepted" if that's alright?

File: .github/workflows/vuln.yml [hunk 1]
```diff
@@ -3,11 +3,30 @@ name: Vulnerability Check
...
+      version-list: ${{ steps.go-dev-stable-versions.outputs.version-list }}
+    steps:
+      - name: List the latest stable versions of Go
+        id: go-dev-stable-versions
+        run: |
+          VERSIONS=$(curl -s https://go.dev/dl/?mode=json | jq -r '.[].version' | sed -e "s/^go//")
+          readarray -t VERSIONS_ARRAY <<< "$VERSIONS"
+          VERSIONS_JSON=$(jq --compact-output --null-input '$ARGS.positional' --args -- "${VERSIONS_ARRAY[@]}")
+          # VERSIONS_JSON should be in a format that fromJson(...) can read
+          echo "version-list=$VERSIONS_JSON" >> $GITHUB_OUTPUT
```
Comment (hajimehoshi) [inline .github/workflows/vuln.yml:19]: What about

```sh
versions_json=$(curl -s https://go.dev/dl/?mode=json | jq '.[].version' | sed -e 's/^"go/"/' | jq -s -c '.')
echo "version-list=$versions_json" >> $GITHUB_OUTPUT
```

File: .github/workflows/vuln.yml [hunk 2]
```diff
@@ -3,11 +3,30 @@ name: Vulnerability Check
...
+
+      - name: Notify on go-dev-stable-versions
+        run: echo "::Notice::version-list is ${{ steps.go-dev-stable-versions.outputs.version-list }}"
```
Comment (hajimehoshi) [inline .github/workflows/vuln.yml:22]: I prefer small caps:

```sh
echo "::notice::version-list is ${{ steps.go-dev-stable-versions.outputs.version-list
```

**Review by hajimehoshi — COMMENTED** [review]
Nice!

**Comment by hajimehoshi** [comment]
> If not specifically avoiding participating in hacktoberfest, could this PR be labelled with "hacktoberfest-accepted" if that's alright?

That's fine! I'm not familiar with Hacktoberfest, but I was wondering if I have to create the label.

File: .github/workflows/vuln.yml (hunk 1 above)
Comment (Skenvy) [reply .github/workflows/vuln.yml]: Swapped to this.

**Comment by hajimehoshi** [comment]
> There's a "for maintainers" section on the site [here](https://hacktoberfest.com/participation/#maintainers) -- but essentially the two opt-in methods are either tagging the repo "hacktoberfest", or individual PRs "hacktoberfest-accepted".

Done (added a label)

File: .github/workflows/vuln.yml [hunk 3]
```diff
@@ -3,11 +3,27 @@ name: Vulnerability Check
...
+      version-list: ${{ steps.go-dev-stable-versions.outputs.version-list }}
+    steps:
+      - name: List the latest stable versions of Go
+        id: go-dev-stable-versions
+        run: |
+          versions_json=$(curl -s https://go.dev/dl/?mode=json | jq '.[].version' | sed -e 's/^"go/"/' | jq -s -c '.')
+          echo "version-list=$versions_json" >> $GITHUB_OUTPUT
+
+      - name: Notify on go-dev-stable-versions
+        run: echo "::notice::version-list is ${{ steps.go-dev-stable-versions.outputs.version-list }}"
```
[the remark this answers is not in the document]
Comment (Skenvy) [reply .github/workflows/vuln.yml:19]: Yea that is a bit weird that it seems to withhold the quotes when printing a context value, but if you look in the step, the instruction it populates it with shows the expected quotes, and fromJson reads it correctly.

**Review by hajimehoshi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**